src/main/resources/application.conf

```diff
@@ -49,6 +49,12 @@ nav {
             discovery_endpoint = ${?LOGINSERVICE_IDPORTEN_DISCOVERY_URL}
             audience = ""
             audience = ${?LOGINSERVICE_IDPORTEN_AUDIENCE}
+        },{
+            alias = "tokenx"
```

Comment:
Er dette i stedet for id-porten? Gå gjennom tokenx

Reply:
Det stemmer. Registrene vi kaller (aareg, breg-proxy) kommer ikke til å støtte id-porten issuer direkte. 
Dermed må apiene våres støtte id-porten, og deretter utveksle id-porten tokenet med tokendings utstedt for ønsket tjeneste.
Flyt blir da som følger: 

1. Bruker logger inn via id-porten 
2. frontend kaller api med idporten token 
3. apiet validerer id-porten
4.  apiet utveksler tokenet mot tokendings ment for k9-selvbetjening-oppslag/k9-mellomlagring
5. apiet kaller k9-selvbetjening-oppslag for å hente data
6. k9-selvbetjening-oppslag validerer tokenet
7. k9-selvbetjening-oppslag kaller tokendings for å utveksle tokenet for å få en token ment for aareg/breg-proxy

Reply:
Nice! Høres jo greit ut :P Legger vekslelogikken inn i dusseldorf-ktor, så vi bare kan gjenbruke det?

Reply:
Veldig mye av vekslelogikken ligger allerede i dusseldorf-ktor.
Man må bare konfigurere opp en client og kalle tokendings med audience på tjenesten man ønsker token for. Vi gjør det for kall mot PDL idag :) 
https://github.com/navikt/k9-selvbetjening-oppslag/blob/master/src/main/kotlin/no/nav/k9/utgaende/gateway/PDLProxyGateway.kt#L32

Reply:
🪄